src/app/modules/verification/verification.service.ts

```diff
@@ -452,18 +457,32 @@ const sendOtpForField = (
       return fieldId
         ? FormService.retrieveFormById(formId)
             // check if we should allow public user to request for otp
+            .andThen((form) => {
+              return okAsync(form)
+            })
             .andThen((form) =>
-              shouldGenerateMobileOtp(form, fieldId, recipient),
-            )
-            // call sms - it should validate the recipient
-            .andThen(() =>
-              SmsFactory.sendVerificationOtp(
-                recipient,
-                otp,
-                otpPrefix,
-                formId,
-                senderIp,
-              ),
+              shouldGenerateMobileOtp(form, fieldId, recipient).andThen(() => {
+                const postmanFlag = verifyUserBetaflag(
```

Comment:
As discussed last Friday, we might not need to rely on beta flags. We can perform a try-catch instead. Try block will contain the call to postman and if an error occurs, we catch it by falling back on twilio. We can implement a feature flag in the catch block to turn off the fallback on 1st July. 

Makes it easier for us to properly test that the postman migration is going well ahead of the launch

Reply:
I think I wasn't clear enough last Friday. This flag was for us to run through a dry-run to ensure that our production configurations on both FormSG + Postman are configured properly. Consequence of incorrect configurations would be sending with the incorrect SenderID (i.e., OGP instead of FormSG).

Correction can be done while in production, but might lead to confusing SMS on the recipient endpoint where their OTP SMSes would be suddenly switch from OGP to FormSG. 

Would prefer to do a controlled small scale test instead of a binary switch.

Reply:
Wouldn't postman configuration issues be caught in UAT if we mirror them with prod? Think it's fine as well but the try-catch will be useful post-prod regardless